src/linear_algebra/affine_space/basis.lean

```diff
@@ -44,26 +44,35 @@ open set
 universes u₁ u₂ u₃ u₄
 
 /-- An affine basis is a family of affine-independent points whose span is the top subspace. -/
+@[protect_proj]
 structure affine_basis (ι : Type u₁) (k : Type u₂) {V : Type u₃} (P : Type u₄)
   [add_comm_group V] [affine_space V P] [ring k] [module k V] :=
-(points : ι → P)
-(ind : affine_independent k points)
-(tot : affine_span k (range points) = ⊤)
+(to_fun : ι → P)
+(ind' : affine_independent k to_fun)
+(tot' : affine_span k (range to_fun) = ⊤)
 
-variables {ι : Type u₁} {k : Type u₂} {V : Type u₃} {P : Type u₄}
-variables [add_comm_group V] [affine_space V P]
+variables {ι ι' k V P : Type*} [add_comm_group V] [affine_space V P]
 
 namespace affine_basis
 
 section ring
 
-variables [ring k] [module k V] (b : affine_basis ι k P)
+variables [ring k] [module k V] (b : affine_basis ι k P) {s : finset ι} {i j : ι} (e : ι ≃ ι')
+include V
+
+instance fun_like : fun_like (affine_basis ι k P) ι (λ _, P) :=
```

Comment:
Note that we don't have this instance for `basis` as far as I can tell.

Reply:
We have [basis.has_coe_to_fun](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basis.html#basis.has_coe_to_fun). Surely it's desirable to upgrade it to `fun_like`?

Reply:
Added